### PR TITLE
Panning

### PR DESCRIPTION
**Pull request by EclipseIsDead**
## Motivation and Context
This is our implementation for dynamic panning, without using the SVG and calculating position differences of the mouse.

## Your Changes

**Description**:
This is our implementation for the panning of the graph SVG, and removing the current panning buttons. I've also updated and removed other obsolete functions and components, like the mouseDown event handlers. 

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring (internal change to codebase, without changing functionality)
- [x] Documentation update (change that modifies or updates documentation only)

## Testing

As of right now, it does what I expect it to do. All states are updated correctly, and as far as I can tell using web browser testing the existing functionality has not been damaged.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CircleCI tests have passed.

### CODE REVIEW

File: js/components/graph/Graph.js [hunk 1]
```diff
@@ -748,12 +786,14 @@ export default class Graph extends React.Component {
...
 
+export {ZOOM_INCREMENT, KEYBOARD_PANNING_INCREMENT}
+
 /** Helper function for parsing hybrid node's text
  *
  * @param {string} s
  * @returns {Array}
  */
-function parseAnd(s) {
+ function parseAnd(s) {
```
Comment (david-yz-liu) [inline js/components/graph/Graph.js:796]: Revert this change and all changes in this file below this line (including the removal of a new line to end the file).

**Review by david-yz-liu — COMMENTED** [review]
@EclipseIsDead @kenneth-miura @leenlab2 great job! I've left comments on this PR since I think it's a merged version of all three of your branches. I'll let the three of you work out how to address these comments.

File: js/components/graph/Graph.js [hunk 2]
```diff
@@ -397,25 +469,45 @@ export default class Graph extends React.Component {
...
+    if (zoomMode === this.zoomEnum.ZOOM_IN){
+      newZoomFactor -= ZOOM_INCREMENT;
+    } else if (zoomMode === this.zoomEnum.ZOOM_OUT) {
+      newZoomFactor += ZOOM_INCREMENT;
+    } else if (zoomMode === this.zoomEnum.RESET_ZOOM){
+      newZoomFactor = 1
+    }
+    let newViewboxWidth = this.state.width;
+    let newViewboxHeight = this.state.height;
+    if (document.getElementById("generateRoot")){
```
[the remark this answers is not in the document]
Comment (kenneth-miura) [reply js/components/graph/Graph.js]: The special case is there so that the graph renders in the middle in **Generate** mode. Since the fix that allows **Generate** mode to run is in a different PR, I can't test any better solutions.

Should we keep the code from the else branch for now, and then fix any issues that arise in **Generate** once the Generate fix PR is merged in, so we can test while developing?

File: js/components/graph/Graph.js [hunk 3]
```diff
@@ -425,73 +517,20 @@ export default class Graph extends React.Component {
...
-      });
-    } else if (direction === "right") {
-      this.setState({
-        horizontalPanFactor: this.state.horizontalPanFactor + panFactorRate
-      });
-    }
-  };
-
   resetZoomAndPan = () => {
+    this.zoomViewbox(this.zoomEnum.RESET_ZOOM);
```
[the remark this answers is not in the document]
Comment (kenneth-miura) [reply js/components/graph/Graph.js:517]: What about setting `zoomFactor` to 1, and resetting `viewBoxDim` to it's original value in this call to `setState` and remove the call to `zoomViewBox`? So that would look like
```
this.setState({
zoomFactor:1,
verticalPanFactor:0,
horizontalPanFactor:0,
viewBoxDim: {width: this.state.width, height: this.state.height}
```

It would mean `zoomViewbox` isn't the only location where the viewbox's dimensions are changed, but I think it is fine. Let me know what you think.

[the remark this answers is not in the document]
Comment (david-yz-liu) [reply js/components/graph/Graph.js:517]: Yes, I think that could work as well.

File: js/components/graph/Graph.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (leenlab2) [reply js/components/graph/Graph.js]: The generate fix has been merged so try pulling now

[the remark this answers is not in the document]
Comment (david-yz-liu) [reply js/components/graph/Graph.js]: Yes that's interesting, I'll revise my original comment. Why don't you always use the logic in the generate mode (taking the max of the width and container width, and same for height), even for the graph page. Hopefully this should make the smaller graphs on the graph page centered as well.

Also, I just pulled in #1216, so if all of you pull from master, you'll get those fixes!

[the remark this answers is not in the document]
Comment (leenlab2) [reply js/components/graph/Graph.js]: I tested this and there is a bug here. As is, in Generate, when you try zooming it "jumps" to a suddenly larger size, and does not return to the original size properly when reset. This is because the initial value of `viewBoxDim.width` is not set to the maximum of data.width and containerWidth, yet it is used anyway on the first render.

[the remark this answers is not in the document]
Comment (kenneth-miura) [reply js/components/graph/Graph.js]: If we always use the logic in generate mode (taking max of width & container width), the graph page also renders in the middle, like so. Is this fine?
image

[the remark this answers is not in the document]
Comment (david-yz-liu) [reply js/components/graph/Graph.js]: @kenneth-miura hmmm yeah, it will also depend on screen size/resolution. There's a deeper issue here, so for now I'm okay leaving the code as-is, with the separate logic for the generate page.

**Comment by david-yz-liu** [comment]
Merged as part of #1213.
